go-controller/pkg/node/gateway_iptables.go

```diff
@@ -467,6 +467,37 @@ func getLocalGatewayNATRules(ifname string, cidr *net.IPNet) []nodeipt.Rule {
 			Protocol: protocol,
 		},
 	}
+	if util.IsNetworkSegmentationSupportEnabled() {
+		// NOTE: Ordering is important here, the RETURN must come before
+		// the MASQUERADE rule. Please don't change the ordering.
+		srcUDNMasqueradePrefix := config.Gateway.V4MasqueradeSubnet
+		defaultNetworkReservedMasqueradePrefix := config.Gateway.MasqueradeIPs.V4HostMasqueradeIP.String() + "/29"
```

Comment:
this is obvio wrong to hardcode /29 and /125.... change this to use the mask of the subnet instead

Reply:
no actually the logic is correct :D its always the first 6 IPs that we want to add the RETURN for.